### DISCUSSION TITLE
Apache Pulsar

### DISCUSSION THREAD

**Discussion by aswinkumaarrs** [apache/pulsar#21624] [Q&A]
Hi,

I'm in the learning phase for Apache Pulsar and I would like to implement a POC which involves 3 machines.
Machine A will only have Producer Client, Machine B will have Pulsar cluster running and Machine C will have Consumer Client.
Is it possible to implement this? 
If it is so, How can we do it?
Kindly guide me what are the steps I need to do!

**Comment by jak78** [comment]
Hi! Welcome to the community!
2 quick questions for me to help you:
- what programming languages will you use for the clients?
- do you use Kubernetes or Docker for this POC?

**Reply by aswinkumaarrs** [reply]
I'm trying to deploy it on Bare Metal and using C# client
Thanks for the reply☺️

**Reply by jak78** [reply]
Thanks

Using a single node for your Pulsar cluster isn't recommended for production environments. You'll need redundancy and replication for that. If your POC is aimed at testing Pulsar's reliability in a production setting, then it's advisable to use multiple nodes. You can follow the guidelines in this tutorial: [Pulsar Bare Metal Deployment](https://pulsar.apache.org/docs/3.1.x/deploy-bare-metal/).

For a simpler setup, one that's not ready for production but is great for getting a feel of Pulsar, you could start with a standalone Pulsar on Machine B. This page offers a good starting point: [Getting Started with Pulsar Standalone](https://pulsar.apache.org/docs/3.1.x/getting-started-standalone/). It's preferable to use a Unix-like OS for Machine B.

To handle message consumption and production on Machines A and C, this page will be helpful: [Pulsar .NET Client Libraries](https://pulsar.apache.org/docs/3.1.x/client-libraries-dotnet/).

**Reply by aswinkumaarrs** [reply]
Thank you so much ☺️.I have implemented successfully.

**Reply by jak78** [reply]
@aswinkumaarrs I'm glad you implemented it successfully.
